File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/BottomActionBarView.swift
import SwiftUI

struct BottomActionBarView: View {
    let price: Double
    var onAddToCart: () -> Void
    var onBuyNow: () -> Void
    var onOpenCart: () -> Void

    var body: some View {
        HStack(spacing: 12) {
            // 1. Cart Icon Button (Square outline)
            Button(action: {
                HapticManager.shared.selection()
                onOpenCart()
            }) {
                Image(systemName: "cart")
                    .font(.system(size: 20))  // Slightly larger icon
                    .foregroundColor(Color(hex: "#111827"))
                    .frame(width: 50, height: 50)
                    .background(Color.white)
                    .cornerRadius(8)
                    .overlay(
                        RoundedRectangle(cornerRadius: 8)
                            .stroke(Color(hex: "#E5E7EB"), lineWidth: 1)
                    )
            }

            // 2. Add to Cart Button (Outline)
            Button(action: {
                HapticManager.shared.impact(style: .heavy)
                onAddToCart()
            }) {
                Text("Add to Cart")
                    .font(.system(size: 15, weight: .bold))
                    .foregroundColor(Color(hex: "#111827"))
                    .frame(maxWidth: .infinity)
                    .frame(height: 50)
                    .background(Color.white)
                    .cornerRadius(8)
                    .overlay(
                        RoundedRectangle(cornerRadius: 8)
                            .stroke(Color(hex: "#E5E7EB"), lineWidth: 1)
                    )
            }

            // 3. Buy Now Button (Yellow, with price)
            Button(action: {
                HapticManager.shared.impact(style: .heavy)
                onBuyNow()
            }) {
                VStack(spacing: 0) {
                    Text("Buy now")
                        .font(.system(size: 15, weight: .black))
                        .foregroundColor(Color(hex: "#111827"))
                    Text("at ₹\(Int(price))")
                        .font(.system(size: 11, weight: .bold))
                        .foregroundColor(Color(hex: "#111827"))
                }
                .frame(maxWidth: .infinity)
                .frame(height: 50)
                .background(Color(hex: "#FACC15"))  // Gold/Yellow
                .cornerRadius(8)
            }
        }
        .padding(12)
        .background(Color.white)
        .shadow(color: Color.black.opacity(0.05), radius: 4, x: 0, y: -2)
    }
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/ui/components/BottomActionBarView.kt
package com.localforvocalstartup.app.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun BottomActionBarView(
    price: Double,
    onAddToCart: () -> Unit,
    onBuyNow: () -> Unit,
    onOpenCart: () -> Unit
) {
    Surface(
        color = Color.White,
        shadowElevation = 8.dp,
        modifier = Modifier.navigationBarsPadding()
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(12.dp),
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Cart Icon Button (Square outline)
            OutlinedButton(
                onClick = onOpenCart,
                modifier = Modifier.size(50.dp),
                shape = RoundedCornerShape(8.dp),
                contentPadding = PaddingValues(0.dp),
                colors = ButtonDefaults.outlinedButtonColors(contentColor = Color(0xFF111827)),
                border = BorderStroke(1.dp, Color(0xFFE5E7EB))
            ) {
                Icon(Icons.Default.ShoppingCart, contentDescription = "Open Cart", modifier = Modifier.size(20.dp))
            }

            // Add to Cart Button
            OutlinedButton(
                onClick = onAddToCart,
                modifier = Modifier
                    .weight(1f)
                    .height(50.dp),
                shape = RoundedCornerShape(8.dp),
                colors = ButtonDefaults.outlinedButtonColors(contentColor = Color(0xFF111827)),
                border = BorderStroke(1.dp, Color(0xFFE5E7EB))
            ) {
                Text("Add to Cart", fontSize = 15.sp, fontWeight = FontWeight.Bold)
            }

            // Buy Now Button
            Button(
                onClick = onBuyNow,
                modifier = Modifier
                    .weight(1f)
                    .height(50.dp),
                shape = RoundedCornerShape(8.dp),
                colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFFACC15)), // Gold/Yellow
                contentPadding = PaddingValues(0.dp)
            ) {
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.Center
                ) {
                    Text("Buy now", fontSize = 15.sp, fontWeight = FontWeight.Black, color = Color(0xFF111827))
                    Text("at ₹${price.toInt()}", fontSize = 11.sp, fontWeight = FontWeight.Bold, color = Color(0xFF111827))
                }
            }
        }
    }
}
